lib/net/http/header.rb

```diff
@@ -126,46 +126,59 @@
 # Various convenience methods retrieve values, set values, query values,
 # set form values, or iterate over fields.
 #
+# === Setters
+#
+# \Method #[]= can set any field, but does little to validate
+# to validate the new value;
+# most of the other setter methods provides some validation:
+#
+# - #[]=: Sets the string or array value for the given key.
+# - #add_field: Creates or adds to the array value for the given kry.
+# - #basic_auth: Sets the string authorization header for <tt>'Authorization'</tt>.
+# - #content_length=: Sets the integer length for field <tt>'Content-Length</tt>.
+# - #content_type=: Sets the string value for field <tt>'Content-Type'</tt>.
+# - #proxy_basic_auth: Sets the string authorization header for <tt>'Proxy-Authorization'</tt>.
+# - #set_range: Sets the value for field <tt>'Range'</tt>.
+#
+# === Form Setters
+#
+# - #set_form: Sets an HTML form data set.
+# - #set_form_data: Sets header fields and a body from HTML form data.
+#
 # === Getters
 #
-# - #[]: Returns the string value for the given field.
+# \Method #[] can retrieve the value of any field that exists,
+# but always as a string;
+# most of the other getter methods return something different
+# from the simple string value:
+#
+# - #[]: Returns the string field value for the given key.
 # - #content_length: Returns the integer value of field <tt>'Content-Length'</tt>.
 # - #content_range: Returns the Range value of field <tt>'Content-Range'</tt>.
 # - #content_type: Returns the string value of field <tt>'Content-Type'</tt>.
+# - #fetch: Returns the string field value for the given key,
+#   or passes it to the block, if given.
```

Comment:
`fetch` returns the default value or the value of the block if it doesn't exist (and doesn't pass the value to the block). I couldn't think of a good way to explain that in the short sentence, so I just copied the sentence from `Hash#fetch`.

```suggestion
# - #fetch: Returns the string field value for the given key.
```